icu4c/source/i18n/unicode/calendar.h

```diff
@@ -2564,7 +2564,6 @@ Calendar::internalSet(UCalendarDateFields field, int32_t value)
     fIsSet[field]     = true; // Remove later
 }
 
-#ifndef U_HIDE_INTERNAL_API
```

Comment:
optional: you might be able to use `#ifndef U_FORCE_HIDE_INTERNAL_API` for the macro, similar to how we use `U_FORCE_HIDE_DRAFT_API` etc. elsewhere ([code search](https://github.com/search?q=repo%3Aunicode-org%2Ficu%20U_FORCE_HIDE_&type=code)) to make the header test happy without breaking normal use of the non-FORCE config

Reply:
I am confused (and still is) why we have the following inside [i18n/unicode/gregocal.h](https://github.com/unicode-org/icu/commit/fbc1f33e7e8afeb92ed7498fca1a1686d8b6b9c4#diff-ded31ae20bc831e7bdac6226b31f22422cf1b5eb5908d08c450d1619561a6155)
```

    /**
     * @return true if this calendar has the notion of a default century
     * @internal 
     */
    virtual UBool haveDefaultCentury() const override;

    /**
     * @return the start of the default century
     * @internal
     */
    virtual UDate defaultCenturyStart() const override;
    /**
     * @return the beginning year of the default century
     * @internal 
     */
    virtual int32_t defaultCenturyStartYear() const override;
```
which was marked by @internal but NOT wrap between 

#ifndef U_HIDE_INTERNAL_API and  #endif

Reply:
Because, if you compile with U_HIDE_INTERNAL_API, the vtable becomes different from the real one exported from the DLL, and then the program would crash. So we have some limitations of where we can use U_HIDE_xxx_API.

We might be able to use U_FORCE_HIDE_INTERNAL_API.